src/main/java/org/matsim/urbanEV/ChargerPlacer.java

```diff
@@ -0,0 +1,185 @@
+package org.matsim.urbanEV;
+
+import com.google.common.collect.Iterables;
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvException;
+import org.apache.commons.compress.utils.Lists;
+import org.apache.log4j.Logger;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.prep.PreparedGeometry;
+import org.matsim.api.core.v01.Coord;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.Scenario;
+import org.matsim.api.core.v01.TransportMode;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.api.core.v01.population.*;
+import org.matsim.contrib.ev.infrastructure.Charger;
+import org.matsim.contrib.ev.infrastructure.ChargerSpecification;
+import org.matsim.contrib.ev.infrastructure.ChargerWriter;
+import org.matsim.contrib.ev.infrastructure.ImmutableChargerSpecification;
+import org.matsim.core.config.Config;
+import org.matsim.core.network.NetworkUtils;
+import org.matsim.core.network.algorithms.TransportModeNetworkFilter;
+import org.matsim.core.network.io.MatsimNetworkReader;
+import org.matsim.core.network.io.NetworkWriter;
+import org.matsim.core.population.PopulationUtils;
+import org.matsim.core.router.TripStructureUtils;
+import org.matsim.core.utils.collections.Tuple;
+import org.matsim.core.utils.geometry.GeometryUtils;
+import org.matsim.core.utils.gis.ShapeFileReader;
+import org.matsim.run.RunBerlinScenario;
+import org.matsim.utils.gis.shp2matsim.ShpGeometryUtils;
+import org.matsim.withinday.utils.EditPlans;
+import org.opengis.feature.simple.SimpleFeature;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+public class ChargerPlacer {
```

Comment:
please rename into ChargerAssignment or something similar. @Jonas116 Please clean up the code